spec/classes/pulpcore_spec.rb

```diff
@@ -51,6 +51,29 @@
           is_expected.to contain_exec('pulpcore-manager reset-admin-password --random')
         end
 
+        it 'does not install Redis' do
+          is_expected.not_to contain_class('redis')
+        end
+
+        it 'does not configure Pulpcore connection to Redis' do
+          is_expected.to contain_concat__fragment('base')
+            .without_content(/REDIS_HOST/)
+            .without_content(/REDIS_POST/)
+            .without_content(/REDIS_DB/)
+        end
+
+        it 'does not configure content caching' do
+          is_expected.to contain_concat__fragment('base')
+            .without_content(/CACHE_ENABLED = True/)
+            .without_content(%r{CACHE_SETTINGS = \{\n    'EXPIRES_TTL': 60,\n\}})
+        end
+
+        it 'configures pulpcore to use PostgreSQL tasking system' do
+          is_expected.to contain_concat__fragment('base')
+            .with_content(/USE_NEW_WORKER_TYPE = True/)
+          is_expected.to contain_systemd__unit_file('pulpcore-resource-manager.service').with_ensure('absent')
+        end
```

Comment:
This are OK here, but they feel better up here https://github.com/theforeman/puppet-pulpcore/pull/207/files#diff-257fe3136ecfdbfcb123b6445ff2de6c21d1fe08b0d6af54c08dde40637d2594R19 for me as it connects all of the configuration file together.

Reply:
I started to think that it makes sense to structure the tests around the real world use case rather than the specific object being tested..

So instead of having all tests for the configuration file in the same place, have a test for 'content caching is disabled', 'postgresql tasking system is used', and so on.

I see two primary benefits: 1. This feels more maintainable in the longer term (when we make modifications to a specific feature, the tests for that feature are all located in that block, rather than spread out across different components the feature touches). 2. This should lead to more readable test output, and particularly failures. So instead of getting a failure like `default parameters - it configures pulpcore - it is expected to contain a file with (lots of stuff, including all kinds of things that aren't relevant to the specific thing you broke)` you would instead see something more like `default parameters - it uses new tasking system - it is expected to contain a configuration file with specific things that are relevant to the tasking system`

I want to pursue this idea more consistently in my planned refactoring. For this PR I thought, it makes sense to write the net-new tests using this style, to minimize the amount of re-writing I have to do later.